### PR TITLE
switch to using typelib's value and valuevisitor types so we get the …

### PR DESCRIPTION
**Pull request by vbargsten**
…resulting data directly without further raw data parsing;
fix empty range selection;
add progress output

This should fix issue #6 by using typelib's load function.
We cannot cast directly log data to types, the data has to be loaded since the binary representation may differ (such as for vectors). This also removes a lot of duplicate code.
I'm not 100% sure the init of the typelib value is supposed to be done like that,
but it works and valgrind doesn't complain.

### CODE REVIEW

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:277]: Can't compile this:

```
In file included from /usr/include/c++/9/cassert:44,
                 from /home/afabisch/Projekte/pocolog2msgpack_test/tools/pocolog2msgpack/src/Converter.cpp:14:
/home/afabisch/Projekte/pocolog2msgpack_test/tools/pocolog2msgpack/src/Converter.cpp: In destructor ‘virtual Converter::~Converter()’:
/home/afabisch/Projekte/pocolog2msgpack_test/tools/pocolog2msgpack/src/Converter.cpp:277:13: error: ‘data’ was not declared in this scope
  277 |     assert(!data);
```

I guess you build with `-DNDEBUG` (release mode?). Delete this line if you don't need the `data` anymore.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:130]: I'd like to keep the coding style consistent in this repository: move `{` to the beginning of the next line.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:137]: Please remove trailing whitespaces.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:203]: A new feature. Should we document this somewhere?

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:270]: Why does this need brackets?

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:156]: To keep a consistent style throughout this repository, please also use camel case, not snake case.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:338]: Cosmetic change: add whitespace before `<<`.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:346]: ... in all of these functions.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:416]: Trailing whitespaces.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:418]: This should be bool right? I think in this case we can write it down explicitly here. Otherwise we can cast it to bool to avoid confusing implicit casting.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:448]: Cosmetic: whitespace before `<<`

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:453]: I think a warning message would be better than an error. Maybe this is intended behavior to cut the converted log.

File: src/Converter.cpp
Comment (AlexanderFabisch) [inline src/Converter.cpp:185]: Coding style: please remove whitespaces around `f`.

**Review by AlexanderFabisch — CHANGES_REQUESTED** [review]
The converted data from @Rauldg 's short log look good to me. Unit tests still pass, which indicates that nothing broke so far. Apart from some cosmetic changes this PR looks good to me. Great contribution that significantly improves the package! I would still wait for @Rauldg 's feedback though.

**Comment by AlexanderFabisch** [comment]
There are currently some dependency issues that I have to solve before I can merge this PR. The new version of msgpack for Python is not backwards compatible. I would like to switch to `pip install msgpack` to avoid problems. This would, however, not be compatible to Python 2 anymore.

**Comment by AlexanderFabisch** [comment]
Btw. is the option `size` still required? If not, can you remove it from the code (and the readme)?

File: src/Converter.cpp
Comment (vbargsten) [reply src/Converter.cpp:270]: For clarity.

File: src/Converter.cpp
Comment (vbargsten) [reply src/Converter.cpp]: I would not call it a feature...but maybe adapt the output or verbosity level if you prefer (the stream names and number of samples are also printed by default).

**Comment by vbargsten** [comment]
> There are currently some dependency issues that I have to solve before I can merge this PR. The new version of msgpack for Python is not backwards compatible. I would like to switch to `pip install msgpack` to avoid problems. This would, however, not be compatible to Python 2 anymore. I'll probably also have to revise the docker image for CI.

How does it relate to this PR? This PR only fixes the conversion, not the format.

**Comment by vbargsten** [comment]
> Btw. is the option `size` still required? If not, can you remove it from the code (and the readme)?

Removed.

**Comment by vbargsten** [comment]
Resolved points addressed in the commits.

File: src/Converter.cpp
Comment (AlexanderFabisch) [reply src/Converter.cpp]: In my opinion it should be quite clear that this is a boolean expression with a boolean return value either with or without brackets. Only complicated expressions with multiple operators and complicated execution order would justify adding brackets for clarity. It's a minor issue though and I can live with that.

**Comment by AlexanderFabisch** [comment]
> > There are currently some dependency issues that I have to solve before I can merge this PR. The new version of msgpack for Python is not backwards compatible. I would like to switch to `pip install msgpack` to avoid problems. This would, however, not be compatible to Python 2 anymore. I'll probably also have to revise the docker image for CI.
> 
> How does it relate to this PR? This PR only fixes the conversion, not the format.

Fair point. I couldn't test it on my computer with Ubuntu 20.04 but as long as CI succeeds it shouldn't be a problem.

**Comment by AlexanderFabisch** [comment]
Thanks for fixing this @vbargsten . I merged this to master now. I'm still waiting for @Rauldg 's final feedback though.

**Pull request merged**
